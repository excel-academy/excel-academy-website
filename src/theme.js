import { themeColor } from '../site-config';

const purples = ['#7b79b0', '#6f6da5', '#eeeefc', '#1c2070'];
const greens = ['#d2eced', '#132c2d'];
const fontSizes = [12, 14, 16, 18, 20, 24, 30, 38, 62];
const space = [0, '2rem', '3rem', '4rem', '5rem', '7rem', '32rem', '40rem', '48rem', '56rem', '64rem'];
export const breakpoints = {
  sm: 0,
  mobile: '30em',
  tablet: '46em',
  desktop: '61.25em',
  wide: '80em',
  extrawide: '105em',
};

export default {
  space,
  fontSizes,
  breakpoints,
  maxWidths: ['48em', '60em', '80em'],
  colors: {
    theme: themeColor,
    purples,
    greens,
  },
  buttons: {
    primary: {
      color: '#fff',
      backgroundColor: themeColor,
      fontSize: fontSizes[4],
      borderColor: '#fff',
      paddingRight: space[4],
      paddingLeft: space[4],
      '&:hover': {
        backgroundColor: '#000',
      },
    },
    outline: {
      color: themeColor,
      backgroundColor: 'transparent',
      boxShadow: 'inset 0 0 2px',
    },
  },
};
