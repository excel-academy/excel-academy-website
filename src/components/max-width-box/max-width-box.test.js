import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from '../../theme';
import MaxWidthBox from './max-width-box-css';

describe('MaxWidthBox', () => {
  it('default renders correctly', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <MaxWidthBox maxWidth={1}>
          <h1>hello world</h1>
        </MaxWidthBox>
      </ThemeProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('styled system renders correctly', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <MaxWidthBox
          css={{
            display: 'flex',
            flexWrap: 'wrap',
            justifyContent: 'center',
            alignItems: 'center',
          }}
          maxWidth={2}
          px={{ sm: 1, tablet: 3 }}
          py={3}
          bg="white"
        >
          <h1>hello world</h1>
        </MaxWidthBox>
      </ThemeProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
