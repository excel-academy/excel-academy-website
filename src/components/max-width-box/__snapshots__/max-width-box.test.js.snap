// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaxWidthBox default renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  max-width: 60em;
  margin: 0 auto;
}

<div
    class="c0"
  >
    <h1>
      hello world
    </h1>
  </div>
</DocumentFragment>
`;

exports[`MaxWidthBox styled system renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  padding-top: 4rem;
  padding-bottom: 4rem;
  padding-left: 2rem;
  padding-right: 2rem;
  background-color: white;
  max-width: 80em;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media screen and (min-width:46em) {
  .c0 {
    padding-left: 4rem;
    padding-right: 4rem;
  }
}

<div
    class="c0"
  >
    <h1>
      hello world
    </h1>
  </div>
</DocumentFragment>
`;
